MobileTraining/MobileTraining/Views/Navigation/NavigationView.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  NavigationView.swift
+//  MobileTraining
+//
+//  Created by Svetoslav Donkov on 14.05.23.
+//
+
+import SwiftUI
+import Factory
+
+struct AppNavigationView: View {
+    let loginRepository = Container().loginRepository()
+    let productRepository = Container().productRepository()
+    let networkManager = Container().networkManager()
+    @State private var isLoggedIn = false;
+    
+    var body: some View {
+        NavigationView {
+            if isLoggedIn {
+                ProductView(productRepository: productRepository)
+            } else {
+                LoginView(isLoggedIn: $isLoggedIn, loginRepository: loginRepository, networkManager: networkManager)
```

Comment:
You can inject the repositories in the viewmodel directly. The DI is exactly for that -- to get instances of the data you need where/when you need them.